### PR TITLE
Feat/masterdata for 2x

### PR DESCRIPTION
**Pull request by pscruzzz**
#### What is the purpose of this pull request?
Extend the functionalities of masterdataFor. More specifically, this PR adds different behaviours to which schema it will choose depending on which masterdata builder version you are using!

If you are using the builder 1.x nothing should change, it will always get the schema that has the same version as the app.

If you are using 2.x, the masterdataFor will fetch which schema has relationship with your current app version before making any document interaction.

#### What problem is this solving?
This PR allows your IO service to connect with masterdata builder 2.x! Learn more about this major version at its [PR](https://github.com/vtex/builder-hub/pull/1149)!

#### How should this be manually tested?
- Clone this branch locally
- Follow the steps described on 'Contributing', it will help you connect your local lib to your IO Service!
- You could also run the unit tests developed in this PR.

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [x] Update `CHANGELOG.md`

### CODE REVIEW

**Review by anamaiabap — APPROVED** [review]

**Comment by fabianaferreira** [comment]
@pscruzzz, I've just seen that the lint check failed. Can you run `yarn format`, and push the changes? Thanks!

**Comment by wagnerlduarte** [comment]
> @pscruzzz, I've just seen that the lint check failed. Can you run `yarn format`, and push the changes? Thanks!

@fabianaferreira I fixed the files and now all checks are running 😄

**Review by beatrizmaselli — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -77,6 +77,8 @@ const ContractsClient = vbaseFor<string, MyContractType>('contracts')
 export type ContractsClient = InstanceType<typeof Contracts>
 ```
 
+> 🚨 Warning: If you are NOT using the version 1.x of Masterdata builder, you should call masterDataFor passing the major version being used as the third parameter ```masterDataFor<MyBookType>('books', undefined, 2)```.
```
Comment (fabianaferreira) [inline README.md:82]: AFAIK, this PR is a breaking change, right? For those who use MD v1

**Review by fabianaferreira — APPROVED** [review]
Thanks for your contribution, @pscruzzz and @wagnerlduarte! 🚀 

My comments are:
- I know you implemented unit tests, but have you tested them in an IO app?
- You adapted the MD that we had, moving it to V1 files, but as far as I could understand, is it going to be a breaking change compared to the previous package version? And if so, how do we plan to let the users know?
- Nit: I'd like to see a more detailed changelog, and more information on the readme file on how to use both.

File: README.md (hunk 1 above)
Comment (beatrizmaselli) [reply README.md:82]: Hello @fabianaferreira !

It's not a breaking change :) 
The builder 1.x will work as it works now and those changes of implementation are only for everybody that will use the 2.x builder. 
Those 2 parameters are not mandatory!

Comment (fabianaferreira) [reply README.md:82]: The fallback is the v1, right?

Comment (beatrizmaselli) [reply README.md:82]: yes!

**Comment by beatrizmaselli** [comment]
> I know you implemented unit tests, but have you tested them in an IO app? 

 Yes, we tested using a service app with the instructions we added at the readme file.
 

>  You adapted the MD that we had, moving it to V1 files, but as far as I could understand, is it going to be a breaking change compared to the previous package version? And if so, how do we plan to let the users know?

 It is not a breaking change because of the way it was implemented, not making the parameter mandatory. Builder 1.x will continue to work in current implementations without any modifications and whoever wants to change the version must update the manifest and add the new parameter.
 

>   Nit: I'd like to see a more detailed changelog, and more information on the readme file on how to use both.

The only change from version 2.x to 1.x is the additional parameter that we include since the changes are more related to the schemas as exemplified in the flowchart we created:
https://whimsical.com/masterdata-builder-v2-9KCMchsbdFCtd9eBBjiiNp
basically, the io clients that instantiated the masterdataFor will continue to do this, but as an invisible cover it is actually instantiating the implemented decorator.
Are there any specific details you miss in ReadMe?
